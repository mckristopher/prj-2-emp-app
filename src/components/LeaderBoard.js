import { connect } from "react-redux"
import '../css/leaderboard.css';

function LeaderBoard(props) {
    return (
        <div className="container">
            <section id="leaderboard">
                <nav className="ladder-nav">
                    <div className="ladder-title">
                    <h1>Standings</h1>
                    </div>
                </nav>
                <table id="rankings" className="leaderboard-results" width="100%">
                    <thead>
                        <tr className="font-20">
                            <th>Rank</th>
                            <th>Users</th>
                            <th>Answered</th>
                            <th>Asked</th>
                        </tr>
                    </thead>
                    <tbody>
                        {
                            props.leaders.map((leader, i) => (
                                <tr className="font-20">
                                    <td>{ i + 1 }</td>
                                    <td>
                                        <div className="leader-info">
                                            <img src={ leader.avatar } className="leader-avatar"/>
                                            <div>
                                                <p><strong>{leader.name}</strong></p>
                                                <p><em>{leader.id}</em></p>
                                            </div>
                                        </div>
                                    </td>
                                    <td>{leader.asked}</td>
                                    <td>{leader.answered}</td>
                                </tr>
                            ))
                        }
                        
                    </tbody>
                </table>
            </section>
        </div>
    )
}

const mapStateToProps = ({ users }) => {
    let leaders = Object.keys(users).map((userId) => {
        let user = users[userId];
        return {
            id: user.id,
            name: user.name,
            avatar: user.avatarURL,
            asked: user.questions.length,
            answered: Object.keys(user.answers).length
        }
    }).sort((a, b) => (b.asked + b.answered) - (a.asked + a.answered));
    return {
        leaders
    };
}

export default connect(mapStateToProps)(LeaderBoard);