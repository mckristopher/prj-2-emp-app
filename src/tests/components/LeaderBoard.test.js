import { render } from '@testing-library/react';

import { Provider } from 'react-redux';
import { BrowserRouter } from "react-router-dom";
import LeaderBoard from '../../components/LeaderBoard';

import store from './mock-store';

describe('LeaderBoard', () => {
    it('should match snapshot', () => {

        let component = render (
            <Provider store={store}>
                <BrowserRouter>
                    <LeaderBoard />
                </BrowserRouter>
            </Provider>
        )
        expect(component).toMatchSnapshot();
    })


});