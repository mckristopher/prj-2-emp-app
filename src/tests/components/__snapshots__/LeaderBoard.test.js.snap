// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderBoard should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <section
          id="leaderboard"
        >
          <nav
            class="ladder-nav"
          >
            <div
              class="ladder-title"
            >
              <h1>
                Standings
              </h1>
            </div>
          </nav>
          <table
            class="leaderboard-results"
            id="rankings"
            width="100%"
          >
            <thead>
              <tr
                class="font-20"
              >
                <th>
                  Rank
                </th>
                <th>
                  Users
                </th>
                <th>
                  Answered
                </th>
                <th>
                  Asked
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="font-20"
              >
                <td>
                  1
                </td>
                <td>
                  <div
                    class="leader-info"
                  >
                    <img
                      alt="Avatar"
                      class="leader-avatar"
                      src="/images/snow.jpg"
                    />
                    <div>
                      <p>
                        <strong>
                          Sarah Edo
                        </strong>
                      </p>
                      <p>
                        <em>
                          sarahedo
                        </em>
                      </p>
                    </div>
                  </div>
                </td>
                <td>
                  4
                </td>
                <td>
                  2
                </td>
              </tr>
              <tr
                class="font-20"
              >
                <td>
                  2
                </td>
                <td>
                  <div
                    class="leader-info"
                  >
                    <img
                      alt="Avatar"
                      class="leader-avatar"
                      src="/images/leaf.jpg"
                    />
                    <div>
                      <p>
                        <strong>
                          John Doe
                        </strong>
                      </p>
                      <p>
                        <em>
                          johndoe
                        </em>
                      </p>
                    </div>
                  </div>
                </td>
                <td>
                  3
                </td>
                <td>
                  2
                </td>
              </tr>
              <tr
                class="font-20"
              >
                <td>
                  3
                </td>
                <td>
                  <div
                    class="leader-info"
                  >
                    <img
                      alt="Avatar"
                      class="leader-avatar"
                      src="/images/tyler.jpg"
                    />
                    <div>
                      <p>
                        <strong>
                          Tyler McGinnis
                        </strong>
                      </p>
                      <p>
                        <em>
                          tylermcginnis
                        </em>
                      </p>
                    </div>
                  </div>
                </td>
                <td>
                  2
                </td>
                <td>
                  2
                </td>
              </tr>
            </tbody>
          </table>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <section
        id="leaderboard"
      >
        <nav
          class="ladder-nav"
        >
          <div
            class="ladder-title"
          >
            <h1>
              Standings
            </h1>
          </div>
        </nav>
        <table
          class="leaderboard-results"
          id="rankings"
          width="100%"
        >
          <thead>
            <tr
              class="font-20"
            >
              <th>
                Rank
              </th>
              <th>
                Users
              </th>
              <th>
                Answered
              </th>
              <th>
                Asked
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              class="font-20"
            >
              <td>
                1
              </td>
              <td>
                <div
                  class="leader-info"
                >
                  <img
                    alt="Avatar"
                    class="leader-avatar"
                    src="/images/snow.jpg"
                  />
                  <div>
                    <p>
                      <strong>
                        Sarah Edo
                      </strong>
                    </p>
                    <p>
                      <em>
                        sarahedo
                      </em>
                    </p>
                  </div>
                </div>
              </td>
              <td>
                4
              </td>
              <td>
                2
              </td>
            </tr>
            <tr
              class="font-20"
            >
              <td>
                2
              </td>
              <td>
                <div
                  class="leader-info"
                >
                  <img
                    alt="Avatar"
                    class="leader-avatar"
                    src="/images/leaf.jpg"
                  />
                  <div>
                    <p>
                      <strong>
                        John Doe
                      </strong>
                    </p>
                    <p>
                      <em>
                        johndoe
                      </em>
                    </p>
                  </div>
                </div>
              </td>
              <td>
                3
              </td>
              <td>
                2
              </td>
            </tr>
            <tr
              class="font-20"
            >
              <td>
                3
              </td>
              <td>
                <div
                  class="leader-info"
                >
                  <img
                    alt="Avatar"
                    class="leader-avatar"
                    src="/images/tyler.jpg"
                  />
                  <div>
                    <p>
                      <strong>
                        Tyler McGinnis
                      </strong>
                    </p>
                    <p>
                      <em>
                        tylermcginnis
                      </em>
                    </p>
                  </div>
                </div>
              </td>
              <td>
                2
              </td>
              <td>
                2
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
